Xplat/src/main/java/vazkii/botania/common/block/block_entity/CellularBlockEntity.java

```diff
@@ -67,8 +67,8 @@ public void update(Level level) {
 		generation = nextGeneration;
 	}
 
-	public boolean hasPoweredParent(Level level) {
-		return flowerCoords != null && level.getBlockEntity(flowerCoords) instanceof DandelifeonBlockEntity && level.hasNeighborSignal(flowerCoords);
+	public boolean hasActiveParent(DandelifeonBlockEntity dandie) {
+		return flowerCoords != null && dandie.getLevel().getBlockEntity(flowerCoords) instanceof DandelifeonBlockEntity parent && dandie.getLevel().hasNeighborSignal(flowerCoords) && (!dandie.overgrowthBoost || parent.isOnSpecialSoil());
```

Comment:
This or-check at the end is weird to me. It's true if the flower it's being transferred to is not boosted, or if the current flower is boosted. (Did i get that right?) So this returns true when both are boosted or neither are boosted, and false if the old flower is not boosted and the new one is. But if the old flower is boosted and the new one is not, this will still return true, right? Which we don't want, right?

It would also be nice if these variable names better reflected which flower it's coming from and which flower it's going to.

Reply:
This method is always called by the new flower, so we only need to make sure that either this is a normal tick (we aren't in an overgrowth tick), or that the other flower is also enchanted, and therefor will tick as well. This means that adjacent flowers will only share cells on ticks where they both are ticking. If the old flower is boosted and the new one isn't, then the new one won't be ticked, so this will never be called, and thus the return value doesn't matter.